zephyr/src/supp_main.c

```diff
@@ -89,6 +89,8 @@ static void iface_cb(struct net_if *iface, void *user_data)
 
 	/* TODO : make this user configurable*/
 	ifaces[0].ifname = "wlan0";
+
+	os_free(ifname);
```

Comment:
We should have a check early on (line 83) to make sure that ifname is not NULL.

Reply:
os_free can take care of NULL pointer, but added a null check in the print.